### ISSUE TITLE
Transition vers l’optimiseur AdamW pour une meilleure généralisation

### ISSUE THREAD

**Issue by AntoineBendafiSchulmann** [AntoineBendafiSchulmann/doc_detector_pl#1]
Actuellement, le projet utilise l'optimiseur Adam pour entraîner le modèle. Adam intègre la régularisation par "weight decay" directement dans la mise à jour des poids, ce qui peut limiter son efficacité. L'optimiseur AdamW, proposé par PyTorch, corrige cette limitation en traitant la régularisation de manière séparée, permettant ainsi une meilleure généralisation du modèle.
Adopter AdamW pourrait aider à :

- Réduire les risques de surapprentissage (overfitting).
- Améliorer la robustesse du modèle.
